whist_server/services/user_db_service.py

```diff
@@ -36,11 +36,23 @@ def add(cls, user: UserInDb) -> str:
     @classmethod
     def get(cls, username: str) -> UserInDb:
         """
-        Similar to 'get(user_id)', but queries by username instead of user id.
+        Gets the user querying the username.
         :param username: of the user
         :return: the user database object
         """
         user = cls._users.find_one({'username': username})
         if user is None:
             raise UserNotFoundError(username=username)
         return UserInDb(**user)
+
+    @classmethod
+    def get_from_github(cls, username: str) -> UserInDb:
+        """
+        Similar to 'get(username)', but queries by github username instead of application username.
+        :param username: github username of the user
+        :return: the user database object
+        """
+        user = cls._users.find_one({'github_username': username})
```

Comment:
Technically the GitHub username is not unique and we should use the ID.
Is it worth changing?

Reply:
I can have a look into it. Should create too much trouble.

Reply:
> Technically the GitHub username is not unique and we should use the ID. Is it worth changing?

I had a look at the documentation. It seems they are unique: https://docs.github.com/en/site-policy/other-site-policies/github-username-policy

Reply:
They are unique, but usernames can be changed

Reply:
Roger that, I'll check for some sort of ID and will update the identification process. 

Is the rest ok?

Reply:
If it is too much work we can leave it as is

Reply:
Shouldn't be to much trouble.

Reply:
@iTitus I changed the behavior to identify by the ID, but still using the GitHub name during creation.